src/Compiler/Microsoft.AspNetCore.Razor.Language/src/ClassifiedSpanVisitor.cs

```diff
@@ -29,15 +30,10 @@ internal class ClassifiedSpanVisitor : SyntaxWalker
     private BlockKindInternal _currentBlockKind;
     private SyntaxNode? _currentBlock;
 
-    public ClassifiedSpanVisitor(RazorSourceDocument source)
+    public ClassifiedSpanVisitor(RazorSourceDocument source, ImmutableArray<ClassifiedSpanInternal>.Builder spans)
```

Comment:
I know the answer is probably no, but I don't suppose there's a way we could make the ctors private and have static methods to get the classified spans/tag helpers that wrap the pooling and just return an `ImmutableArray<ClassifiedSpan>`?

Reply:
Great idea - I'll do that.